### PR TITLE
Fix dropdown z-index style

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

We noticed `z-index` issue on Firefox and Chrome with the dropdown component and generally were looking to change the behavior from an `:hover` call to action to `click`.

# Changes

- change behavior to open on `click`
- present a backdrop behind dropdown that closes dropdown on `click` and `keypress` (for a11y)
- modify style, notably
   - on hover on trigger outline only
   - animation and background color in trigger once click happened
   - cursor pointer on dropdown link to all space

# Notes

Don't know well the CSS here so @meodai feel free to rewite

# Screenshots

Enregistrement de l’écran 2023-05-10 à 08 08 19

### CODE REVIEW

File: src/frontend/src/styles/main.css [hunk 1]
```diff
@@ -2280,6 +2264,7 @@ a.c-action-list__item {
   padding: calc(var(--rs-dropdown-bezel) * 0.75) 0;
   color: var(--rc-text);
   text-decoration: none;
+  cursor: auto;
```
[the remark this answers is not in the document]
Comment (peterpeterparker) [reply src/frontend/src/styles/main.css:2267]: Try on main or mainnet to move the mouse slowly over the opened dropdown. Your mouse cursor will be displayed as a cursor then pointer then cursor then pointer.

This ensure that it's always displayed the same way.

File: src/frontend/src/styles/main.css [hunk 2]
```diff
@@ -774,6 +774,10 @@ a:hover,
   --bgc: var(--rc-onButton);
 }
 
+:is(.c-dropdown):hover .c-icon--outlined {
```
[the remark this answers is not in the document]
Comment (peterpeterparker) [reply src/frontend/src/styles/main.css:777]: Without the fix, on hover:

- border becomes square - loose the small radius
- overall button becomes 1px smaller visually

issue:

issue

fix:

fix

File: src/frontend/src/styles/main.css (hunk 1 above)
[the remark this answers is not in the document]
Comment (peterpeterparker) [reply src/frontend/src/styles/main.css:2267]: We can set `pointer` or else everywhere as you wish? But we need to fix the cursor blinking

File: src/frontend/src/styles/main.css [hunk 3]
```diff
@@ -2217,23 +2219,6 @@ a.c-action-list__item {
   color: var(--rc-interaction);
 }
 
-.c-dropdown::after {
```
[the remark this answers is not in the document]
Comment (peterpeterparker) [reply src/frontend/src/styles/main.css:2220]: Ah, that white background. It's necessary?

Capture d’écran 2023-05-09 à 13 05 53

File: src/frontend/src/styles/main.css (hunk 2 above)
[the remark this answers is not in the document]
Comment (peterpeterparker) [reply src/frontend/src/styles/main.css]: Discussed and redefined offline.

File: src/frontend/src/styles/main.css (hunk 1 above)
[the remark this answers is not in the document]
Comment (peterpeterparker) [reply src/frontend/src/styles/main.css]: Discussed and redefined offline.

File: src/frontend/src/styles/main.css (hunk 3 above)
[the remark this answers is not in the document]
Comment (peterpeterparker) [reply src/frontend/src/styles/main.css:2220]: Discussed and refined offline.

File: src/frontend/src/utils/keyboard.utils.ts
Comment (nmattia) [inline src/frontend/src/utils/keyboard.utils.ts:1]: Can you stay consistent with variable & module naming?

File: src/frontend/src/flows/manage/settingsDropdown.ts [hunk 4]
```diff
@@ -1,22 +1,42 @@
...
+        />`
+      : undefined
+  );
+
   return html` <div class="c-action-list__action c-dropdown">
+    ${asyncReplace(backdrop)}
     <button
       class="c-dropdown__trigger c-action-list__action"
-      aria-expanded="false"
+      aria-expanded="${asyncReplace(ariaExpanded)}"
```
Comment (nmattia) [inline src/frontend/src/flows/manage/settingsDropdown.ts:35]:
```suggestion
      aria-expanded=${asyncReplace(ariaExpanded)}
```

File: src/frontend/src/flows/manage/settingsDropdown.ts [hunk 5]
```diff
@@ -1,22 +1,42 @@
...
+  const ariaExpanded = new Chan(false);
+  const chasmToggle = () => ariaExpanded.send(!ariaExpanded.latest);
+
+  const backdrop = ariaExpanded.map((expanded) =>
+    expanded
+      ? html`<div
+          class="c-dropdown__backdrop"
+          role="button"
+          tabindex="-1"
+          @click=${() => chasmToggle()}
```
Comment (nmattia) [inline src/frontend/src/flows/manage/settingsDropdown.ts:24]: This should only ever close the dropdown, correct? Would `ariaExpanded.send(false)` make sense?

File: src/frontend/src/flows/manage/settingsDropdown.ts [hunk 6]
```diff
@@ -38,3 +58,6 @@ export const settingsDropdown = ({
     </ul>
   </div>`;
 };
+
+export const settingsDropdown = (params: SettingsDropdown): TemplateResult =>
```
Comment (nmattia) [inline src/frontend/src/flows/manage/settingsDropdown.ts:62]:
```suggestion
const settingsDropdown = (params: SettingsDropdown): TemplateResult =>
```

File: src/frontend/src/flows/manage/settingsDropdown.ts [hunk 7]
```diff
@@ -1,22 +1,42 @@
...
+import { Chan } from "$src/utils/utils";
 import { html, TemplateResult } from "lit-html";
+import { asyncReplace } from "lit-html/directives/async-replace.js";
 
-export const settingsDropdown = ({
-  alias,
-  id,
-  settings,
-}: {
+interface SettingsDropdown {
```
Comment (nmattia) [inline src/frontend/src/flows/manage/settingsDropdown.ts:7]: In other modules we never use `interface`, and as much as possible try to keep then type/function signature inside the function where it's used. Would it make sense to do the same here?

```ts
type SettingsDropdown = Parameters<typeof mkChasm>[0];
```

File: src/frontend/src/flows/manage/settingsDropdown.ts [hunk 8]
```diff
@@ -1,22 +1,42 @@
...
   alias: string;
   id: string;
   settings: { fn: () => void; caption: string; action: string }[];
-}): TemplateResult => {
+}
+
+const mkChasm = ({ alias, id, settings }: SettingsDropdown): TemplateResult => {
```
Comment (nmattia) [inline src/frontend/src/flows/manage/settingsDropdown.ts:13]: `mkChasm` is already used somewhere else, plus this is not really a chasm but a dropdown. Would it make sense to use a single function, `settingsDropdown`? What is the reason for `settingsDropdown = mkChasm`?

File: src/frontend/src/styles/main.css [hunk 9]
```diff
@@ -80,9 +80,9 @@
      * technical implementation than design.
     **/
 
-  --z-backdrop: -1;
   --z-background: 1;
   --z-foreground: 2;
+  --z-backdrop: 5;
```
Comment (nmattia) [inline src/frontend/src/styles/main.css:85]: `--z-backdrop` is also used in modals; will this new value work there too?

**Review by nmattia — COMMENTED** [review]
Thanks! just a couple of comments regarding consistency with the rest of the codebase

**Review by nmattia — APPROVED** [review]
LGTM

**Pull request merged**